yt_dlp/update.py

```diff
@@ -110,49 +118,95 @@ def _sha256_file(path):
 
 
 class Updater:
-    def __init__(self, ydl):
+    def __init__(self, ydl, target=None):
         self.ydl = ydl
+        self._exact = True
+
+        if target is None:
+            target = CHANNEL
+
+        self._target_channel, sep, self._target_tag = target.rpartition('@')
+        # Support for `--update-to stable`
+        # It should become `stable@` and not `@stable`
+        if (not sep) and self._target_tag in UPDATE_SOURCES:
+            self._target_channel = self._target_tag
+            self._target_tag = None
+
+        if not self._target_channel:
+            self._target_channel = CHANNEL
+
+        if not self._target_tag:
+            self._exact = False
+            self._target_tag = 'latest'
+        elif self._target_tag != 'latest':
+            self._target_tag = f'tags/{self._target_tag}'
+
+        self._target_repo = UPDATE_SOURCES.get(self._target_channel)
+
+    def _version_compare(self, a, b):
+        if CHANNEL != self._target_channel:
+            return False
+
+        a, b = version_tuple(a), version_tuple(b)
+        return a == b if self._exact else a >= b
 
     @functools.cached_property
     def _tag(self):
-        if version_tuple(__version__) >= version_tuple(self.latest_version):
-            return 'latest'
+        if self._version_compare(self.current_version, self.latest_version):
+            return self._target_tag
 
-        identifier = f'{detect_variant()} {system_identifier()}'
-        for line in self._download('_update_spec', 'latest').decode().splitlines():
+        identifier = f'{detect_variant()} {self._target_channel} {system_identifier()}'
```

Comment:
identifier should use current channel, not target

Reply:
I don't think it should, if we `--update-to` we need to know what the last compatible version for the target is? `--update-to nightly` from `stable` needs to update to the latest supported version on the `nightly` channel. It will result in an invalid tag if we use the current channel, since it matches that channel which is not a valid tag for `nightly`, so the updater will always fail?